sdk/resourcemover/arm-resourcemover/src/operations/moveCollections.ts

```diff
@@ -0,0 +1,945 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License.
+ *
+ * Code generated by Microsoft (R) AutoRest Code Generator.
+ * Changes may cause incorrect behavior and will be lost if the code is
+ * regenerated.
+ */
+
+import * as msRest from "@azure/ms-rest-js";
+import * as msRestAzure from "@azure/ms-rest-azure-js";
+import * as Models from "../models";
+import * as Mappers from "../models/moveCollectionsMappers";
+import * as Parameters from "../models/parameters";
+import { ResourceMoverServiceAPIContext } from "../resourceMoverServiceAPIContext";
+
+/** Class representing a MoveCollections. */
+export class MoveCollections {
+  private readonly client: ResourceMoverServiceAPIContext;
+
+  /**
+   * Create a MoveCollections.
+   * @param {ResourceMoverServiceAPIContext} client Reference to the service client.
+   */
+  constructor(client: ResourceMoverServiceAPIContext) {
+    this.client = client;
+  }
+
+  /**
+   * Creates or updates a move collection.
+   * @param resourceGroupName The Resource Group Name.
+   * @param moveCollectionName The Move Collection Name.
+   * @param [options] The optional parameters
+   * @returns Promise<Models.MoveCollectionsCreateResponse>
+   */
+  create(resourceGroupName: string, moveCollectionName: string, options?: Models.MoveCollectionsCreateOptionalParams): Promise<Models.MoveCollectionsCreateResponse>;
```

Comment:
This parameter is a required parameter. Why is it being marked as optional?

Reply:
If that's the case, you probably need to mark it as required in your rest api specs https://github.com/Azure/azure-rest-api-specs/blob/master/specification/resourcemover/resource-manager/Microsoft.Migrate/stable/2021-01-01/resourcemovercollection.json#L44